src/us/kbase/workspace/test/workspace/WorkspaceTest.java

```diff
@@ -3568,23 +3554,23 @@ public void saveLargeProvenance() throws Exception {
 			methparams.add(TEXT1000);
 		}
 		Provenance p = new Provenance(foo);
-		p.addAction(new ProvenanceAction().withMethodParameters(methparams));
+		p.addAction(ProvenanceAction.getBuilder().withMethodParameters(methparams).build());
 		ws.saveObjects(foo, prov, Arrays.asList( //should work
 				new WorkspaceSaveObject(getRandomName(), data, SAFE_TYPE1, null, p, false)),
 				getIdFactory());
 		
 		
 		methparams.add(TEXT1000);
 		Provenance p2 = new Provenance(foo);
-		p2.addAction(new ProvenanceAction().withMethodParameters(methparams));
+		p2.addAction(ProvenanceAction.getBuilder().withMethodParameters(methparams).build());
 		try {
 			ws.saveObjects(foo, prov, Arrays.asList(
-					new WorkspaceSaveObject(getRandomName(), data, SAFE_TYPE1, null, p, false)),
+					new WorkspaceSaveObject(getRandomName(), data, SAFE_TYPE1, null, p2, false)),
```

Comment:
This test has had a bug in it for years that should have caused it to fail - it was using the wrong provenance here. The reason the test passes when it shouldn't is because the provenance class is mutable.